src/cargo/util/config/mod.rs

```diff
@@ -742,9 +742,20 @@ impl Config {
                 .unwrap_or(false);
         self.target_dir = cli_target_dir;
 
+        // If nightly features are enabled, allow setting Z-flags from config
+        // using the `unstable` table. Ignore that block otherwise.
         if nightly_features_allowed() {
-            if let Some(val) = self.get::<Option<bool>>("unstable.mtime_on_use")? {
-                self.unstable_flags.mtime_on_use |= val;
+            if let Some(unstable_configs) =
+                self.get::<Option<HashMap<String, String>>>("unstable")?
+            {
+                self.unstable_flags
+                    .update_with_table(&unstable_configs)
+                    .with_context(|| "Invalid [unstable] entry in Cargo config")?;
+
+                // NB. It sucks to parse these twice, but doing it again here
+                //     allows the CLI to override config files for both enabling
+                //     and disabling.
+                self.unstable_flags.parse(unstable_flags)?;
```

Comment:
Could this perhaps move from the beginning of the function to the end to avoid doing this twice?

Reply:
It can, but it'll introduce a couple funky edge cases around bits of the config deserializer that are themselves gated on Zflags. In particular I'm thinking of [`advanced_env`](https://github.com/rust-lang/cargo/blob/master/src/cargo/util/config/de.rs#L68).

Reply:
Ah either way is fine, if it needs to go near the beginning it's gotta go near the beginning, alas!

Reply:
I gave this another go and I think (unfortunately) parsing the cli flags twice is the simplest answer for allowing explicit disables at the command line for flags that are otherwise enabled in config.